Sources/SpeziHealthKitBulkExport/BulkHealthExporter.swift

```diff
@@ -0,0 +1,112 @@
+    public enum ObtainSessionResult<Processor: BatchProcessor> {
+        /// The ``BulkHealthExporter`` created a new session.
+        ///
+        /// This case encapsulates both the newly created session, as well as an `AsyncStream` for getting the individual batch processing results.
+        case newSession(BulkExportSession<Processor>, AsyncStream<Processor.Output>)
+        /// The ``BulkHealthExporter`` already had an existing matching session.
+        case existingSession(BulkExportSession<Processor>)
+    }
+    
+    /// Creates or restores a previously-created Health Bulk Export Session, as identified by `id`.
+    ///
+    /// See the ``BulkHealthExporter`` class documentation for more information.
+    @MainActor
+    public func session<Processor: BatchProcessor>(
+        _ id: BulkExportSessionIdentifier,
+        for exportSampleTypes: SampleTypesCollection,
+        startDate: ExportSessionStartDate,
+        endDate: Date = .now,
+        batchSize: ExportSessionBatchSize = .automatic, // swiftlint:disable:this function_default_parameter_at_end
+        using batchProcessor: Processor,
+        startAutomatically: Bool = true
+    ) async throws -> ObtainSessionResult<Processor> {
+        if let session = sessions.first(where: { $0.sessionId == id }) {
+            guard let session = session as? BulkExportSession<Processor> else {
+                // we found an already-running session with the same id, but a different type
+                throw SessionError.conflictingSessionAlreadyExists
+            }
+            return .existingSession(session)
+        } else {
+            let (stream, continuation) = AsyncStream.makeStream(of: Processor.Output.self)
+            let session = try await BulkExportSession<Processor>(
+                sessionId: id,
+                bulkExporter: self,
+                healthKit: healthKit,
+                sampleTypes: exportSampleTypes,
+                startDate: startDate,
+                endDate: endDate,
+                batchSize: batchSize,
+                localStorage: localStorage,
+                batchProcessor: batchProcessor
+            ) { output in
+                continuation.yield(output)
+            }
+            sessions.append(session)
+            if startAutomatically {
+                session.start()
+            }
+            return .newSession(session, stream)
+        }
+    }
+    
+    /// Deletes the persisted state restoration info for a session created in during previous launch of the app, based on its identifier.
+    ///
+    /// - Note: This function can only be used to delete sessions that have not yet been restored during the current lifetime of the app.
+    ///     Once a session identifier has been passed to ``session(_:for:startDate:endDate:batchSize:using:startAutomatically:)``, it cannot be passed to ``deleteSessionRestorationInfo(for:)`` anymore.
```

Comment:
What is the reason for this? Wouldn't it be great to create a session identifier as a static constant as an extension to `BulkExportSessionIdentifier` to use the dot syntax across the app? What is the core reason for keeping this limitation around?

Reply:
we keep the sessions around once they're created, and even if we remove them from the `BulkHealthExporter`s internal array, we can't prevent the app from retaining a reference and just manually restarting it later on even though the session's underlying `LocalStorage`-managed descriptor would've been deleted.

we could add a mechanism where a session can be put into a non-exitable "terminated" state, which would cause all `start()` calls to become no-ops.

Reply:
I agree that there should be a mechanism to "terminate" a session when it is deleted. IMO it might even be fine for `start` to throw an error in that case, could communicate this to the developer/user.

I think it would be good to be able to use the same identifier in a way that it can be passed to `session...` and `delete` within the lifetime to the app, recreating it would be a thing that is normally not expected for identifiers.